### PR TITLE
Feat:add og image component and add feature flag for og image component

### LINKED ISSUE #1066: Create OG Image Component for shared Articles
## Context

The current codebase has an OG image API that generates images with dynamic data on request. 
We need a component based on the Figma designs to handle this. 
The designs involve layers, images, and gradients, so its a good issue for someone looking to work on CSS skills.

Niall discussed this on the stream, which you can watch [here.](https://www.youtube.com/live/lFvdoPmXfzQ?si=LHWmhY1OTpcF6d_S&t=2864)

Check out the [Figma file.](https://www.figma.com/design/MvNb0ffqPj9kb4MrrRXlsK/Og-Image?node-id=0-1&t=B28Hd3rob8berxqj-1)

## Screenshots

OG Image

## Additional info

The API can be found at
app/api/og/route.tsx

### PR DESCRIPTION
**Pull request by South-IN**
# ✨ Codu Pull Request 💻

Fixes #1066 

## Pull Request details

- Implemented a new Open Graph (OG) Image Component for shared articles.
- Added a feature flag using PostHog to toggle the OG Image Component.
- Created a posthog.ts file to create a posthog client for the server side.

## Any Breaking changes
None

## Associated Screenshots
New image component : 
image

Feature flag demo :

https://github.com/user-attachments/assets/66cd038d-ad2e-40c3-b322-ddfc832049a4

### CODE REVIEW

**Review by NiallJoeMaher — CHANGES_REQUESTED** [review]
This looks so cool! Well done. 🥇 

A few small changes. 
- Addng some padding to the surrounding lines.
- You can delete the author name and article tag for the moment because I forgot I use this for the profile page too and will need to figure out how to separate. 
- You can remove the feature flag for now (since without the article tag and the user it'll be safe to push to production). But major kudos for setting it up and doing the screen recording, it made my life much easier. 
- Add `text-balance` to the title so it flows a little more evenly.

375733569-1a427935-6e32-41a3-876e-fb310a91e35b

**Comment by South-IN** [comment]
Thank you for the feedback.
I'll make the required changes.

**Comment by South-IN** [comment]
## Component after requested changes : 
image

- Please let me know incase of more changes.

**Comment by NiallJoeMaher** [comment]
Just linting issues now so if you run `npm run prettier:fix` you should be good to go

File: app/og/route.tsx [hunk 1]
```diff
@@ -8,7 +7,7 @@ const width = 1200;
 export async function GET(request: Request) {
   try {
     const { searchParams } = new URL(request.url);
-
+    const origin = `${request.headers.get("x-forwarded-proto") || "http"}://${request.headers.get("host")}`;
```
Comment (NiallJoeMaher) [inline app/og/route.tsx:10]: Just before I merge. What exactly does this do? 🤔

Comment (South-IN) [reply app/og/route.tsx:10]: In an open graph image we can't use relative path directly in the src attribute.
So I had to use the actual path to the image.
Example : http://localhost:3000/images/og/<image-name> 
Works when used in the src attribute.

The code takes the request origin which can later be used to get the absolute path to the images.

**Review by NiallJoeMaher — APPROVED** [review]

**Pull request merged**